### DISCUSSION TITLE
使用 gmail 注册获取不到头像

### DISCUSSION THREAD

**Discussion by NichtsHsu** [walinejs/waline#398] [Q&A] [answered]
gmail:

![image](https://user-images.githubusercontent.com/35688135/124134777-c12fc080-dab5-11eb-8e37-ff5bed27524e.png)

qq 邮箱：

![image](https://user-images.githubusercontent.com/35688135/124135186-2aafcf00-dab6-11eb-9870-6776fbbc0e67.png)

gravatar:

![image](https://user-images.githubusercontent.com/35688135/124135278-3d2a0880-dab6-11eb-953a-4a892e85b1e6.png)

不知道是哪里没有配置好。代理也挂着。

**Comment by Mister-Hope** [comment]
镜像站应该是有一定的刷新时间的

**Reply by NichtsHsu** [reply]
> 镜像站应该是有一定的刷新时间的

@Mister-Hope  但是我谷歌账号的gravatar至少有三四年的历史了，反而是QQ邮箱的是刚刚才搞的

**Comment by Mister-Hope** [comment, marked as answer]
我重新看了一下这个问题，这个问题可以看成是一个bug，也可以不看成是一个bug。

```
https://0.gravatar.com/avatar/97b37867c49f6be08298efb4b7787314
```
上述链接是有效的。

请注意谷歌邮箱不区分大小写，且强制要求你在注册的时候全部小写，所以你实际上在注册时候填入了错误的邮箱。你的正确谷歌邮箱名应该是纯小写格式

我并不打算强制小写来获取 gravatar，原因是尽管诸如谷歌，QQ这样的邮箱强制在所有地方显示小写的邮箱名，但实际上接收随意大小写的账户名称并在处理时全部小写 (也就是 z@xxx.com 和 Z@xxx.com 相同)，但 RFC 5321 要求电子邮件区分大小写，且的确有少数邮件提供商允许在特定大小写账户存在的时候提供其他大小写账户的注册，(如在已有z@xxx.com 仍允许注册 Z@xxx.com 并把它们视为不同的邮箱)

我的个人建议是手动修改您的数据库将邮箱变为小写，如果你在其他站点评论，请重新使用小写邮箱进行评论。

**Reply by NichtsHsu** [reply]
确实是一个大坑

**Reply by Mister-Hope** [reply]
相关提示已经加入到文档中。

https://github.com/walinejs/waline/commit/7870af0156340c55b7a9d3385484e49dd847e43d
